src/main/java/com/health/threadpool/TaskProcessingService.java

```diff
@@ -326,7 +326,8 @@ public void addUpdateDeleteTutorial(Tutorial tutorial, String requestType) {
             outlinePath = Optional.of(tutorial.getOutlinePath());
         }
 
-        if (requestType.equals(CommonData.ADD_DOCUMENT) && !tutorial.isAddedQueue() && !sm_url.isEmpty()) {
+        if (requestType.equals(CommonData.ADD_DOCUMENT) && !tutorial.isAddedQueue() && !sm_url.isEmpty()
+                && isURLWorking(documentPathforOriginalScript)) {
```

Comment:
don't call isURLWorking twice for the same URL. assign to a boolean variable once and use it.